vip-api/src/main/java/fr/insalyon/creatis/vip/api/security/oidc/OidcConfig.java

```diff
@@ -0,0 +1,86 @@
+package fr.insalyon.creatis.vip.api.security.oidc;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.env.Environment;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.io.FileInputStream;
+import java.net.URI;
+
+import fr.insalyon.creatis.vip.api.CarminProperties;
+
+@Configuration
+public class OidcConfig {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Environment env;
+    private final Collection<OidcServer> servers;
+
+    static private class OidcServer {
+        public String issuer;
+        public String resource;
+
+        OidcServer(String issuer, String resource) {
+            this.issuer = issuer;
+            this.resource = resource;
+        }
+    }
+
+    @Autowired
+    public OidcConfig(Environment env, Resource vipConfigFolder) throws Exception {
+        this.env = env;
+        // Build the list of OIDC servers from config file. If OIDC is disabled, just create an empty list.
+        ArrayList<OidcServer> servers = new ArrayList<>();
+        if (isOIDCActive()) {
+            final String basename = "keycloak.json";
+            try {
+                // read and parse keycloak.json file to get OIDC server URL and clientId
+                String filename = vipConfigFolder.getFile().getAbsoluteFile() + "/" + basename;
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode node = mapper.readTree(new FileInputStream(filename));
+                String baseURL = node.get("auth-server-url").asText();
+                String realm = node.get("realm").asText();
+                String resource = node.get("resource").asText();
+                // Build OIDC server URL from auth-server-url + realm name (this is Keycloak-specific).
+                // We use URI.resolve() instead of just concatenation, to correctly handle optional '/' at the end of baseURL.
+                URI url = new URI(baseURL).resolve("realms/" + realm);
+                servers.add(new OidcServer(url.toASCIIString(), resource));
+            } catch (Exception exception) {
```

Comment:
Not a good practice to have an umbrella catch like that.
About checked exception (aka non-runtime, aka the ones we must deal with), we must declare them at the method signature. We can catch them to log and rethrow then, but it is optional as java exception are pretty detailled.
About RuntimeException, let them bubble up. We can Catch a few if we expect them.
About NPE (NullPointerException), we should prevent them from happening in the code.